src/CoreLib/distributedfunction.fs

```diff
@@ -343,6 +343,115 @@ type internal DistributedFunctionHolder(name:string, capacity:int, executor: (Gu
+                let semaTask = semaphore.WaitAsync(useTimeBudget, chainedCTS.Token )
+                let wrapperExecute (taskEnter:Task<bool>) = 
+                    let elapse = int (( DateTime.UtcNow.Ticks - launchTicks ) / TimeSpan.TicksPerMillisecond )
+                    if taskEnter.Result then 
+                        // Entered
+                        try 
+                            if useTimeBudget > elapse then 
+                                if not chainedCTS.Token.IsCancellationRequested then 
+                                    executor( jobID, useTimeBudget - elapse, input, chainedCTS.Token, observer )
+                                else
+                                    // Operation cancelled. 
+                                    observer.OnCompleted()
+                            else
+                                let ex = TimeoutException( sprintf "ExecuteWithTimebudget: Function %s, time budget %d ms has been exhausted (%d ms)" name timeBudget elapse)
+                                observer.OnError(ex)
+                            chainedCTS.Dispose()
+                            semaphore.Release() |> ignore
+                        with 
+                        | ex -> 
+                            Logger.LogF( LogLevel.Warning, fun _ -> sprintf "ExecuteWithTimebudget, Functioon %s, exception in main loop of %A" name ex )
+                            observer.OnError( ex )
+                            chainedCTS.Dispose()
+                            semaphore.Release() |> ignore
+                    else
+                        try 
+                            if useTimeBudget <= elapse then 
+                                let ex = OperationCanceledException(sprintf "ExecuteWithTimebudget: Function %s, semaphore task is entered as cancelled" name)
+                                observer.OnError(ex)
+                            else
+                                let ex = TimeoutException( sprintf "ExecuteWithTimebudget: Function %s, semaphore task is entered in %d ms(budget %d ms)" name elapse timeBudget )
+                                observer.OnError(ex)
+                            chainedCTS.Dispose()
+                            semaphore.Release() |> ignore 
+                        with
+                        | ex -> 
+                            Logger.LogF( LogLevel.Warning, fun _ -> sprintf "ExecuteWithTimebudget, Functioon %s, exception while not enter, of %A" name ex )
+                            observer.OnError( ex )
+                            chainedCTS.Dispose()
+                let cancelledExecute (taskEnter:Task<bool>) = 
+                    try 
+                        chainedCTS.Dispose()
+                        semaphore.Release() |> ignore 
+                    with
+                    | ex -> 
+                        Logger.LogF( LogLevel.Warning, fun _ -> sprintf "ExecuteWithTimebudget, Functioon %s, exception when dispose of %A" name ex )
```

Comment:
If an exception has filed in disposing chainedCTS.Dispose()  and release semaphore.Release(), there is no point in trying to disposing again (2nd exception?)